codec-socks/src/main/java/io/netty/handler/codec/socksx/v5/DefaultSocks5CommandRequest.java

```diff
@@ -63,7 +57,7 @@ public DefaultSocks5CommandRequest(
             throw new IllegalArgumentException("dstPort: " + dstPort + " (expected: 0~65535)");
         }
 
-        this.type = type;
+        this.type = ObjectUtil.checkNotNull(type, "type");
```

Comment:
lets move the check and assignment before the other stuff to keep the behaviour the same as before.